frappe/integrations/doctype/google_calendar/google_calendar.py

```diff
@@ -682,6 +730,44 @@ def get_recurrence_parameters(recurrence):
 	return frequency, until, byday
 
 
+def get_conference_data(doc):
+	return {
+		"createRequest": {"requestId": doc.name, "conferenceSolutionKey": {"type": "hangoutsMeet"}},
+		"notes": doc.description,
+	}
+
+
+def get_attendees(doc):
+	"""
+	Returns a list of dicts with attendee emails
+	attendee emails are fetched from documents linked (dynamic) in event_participants table
+	only checks for fieldnames 'user', 'user_id', 'email' to find email ids
+	TODO: also check fields of fieldtype Data with options Email
+	"""
+	attendees = []
+
+	for participant in doc.event_participants:
+
+		participant_doc = frappe.get_doc(participant.reference_doctype, participant.reference_docname)
+
+		if participant_doc.meta.has_field("user") and participant_doc.user:
+			attendees.append({"email": participant_doc.user})
+		elif participant_doc.meta.has_field("user_id") and participant_doc.user_id:
+			attendees.append({"email": participant_doc.user_id})
+		elif participant_doc.meta.has_field("email") and participant_doc.email:
+			attendees.append({"email": participant_doc.email})
```

Comment:
yeah, this even i was doubtful.. it would be better we allow user to select email field in the participants table - will add this

Reply:
discussed this with @ankush, maybe just do this when someone has linked contact document(s) ?

Reply:
Yes, Contact document implements linking contact info with arbitrary documents. 

Lets add another column and fetch primary email from the linked contract instead of user directly?  😄

Reply:
thanks! let me try and add that

Reply:
Hi @phot0n I've made the suggested changes. as you get the time, please take a look and let me know?